internal/cmd/argus/instance/update/update.go

```diff
@@ -0,0 +1,217 @@
+			if err != nil {
+				return err
+			}
+
+			instanceLabel, err := argusUtils.GetInstanceName(ctx, apiClient, model.InstanceId, model.ProjectId)
+			if err != nil || instanceLabel == "" {
+				instanceLabel = model.InstanceId
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to update instance %q?", instanceLabel)
+				err = confirm.PromptForConfirmation(cmd, prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req, err := buildRequest(ctx, model, apiClient)
+			if err != nil {
+				var argusInvalidPlanError *cliErr.ArgusInvalidPlanError
+				if !errors.As(err, &argusInvalidPlanError) {
+					return fmt.Errorf("build Argus instance update request: %w", err)
+				}
+				return err
+			}
+
+			_, err = req.Execute()
+			if err != nil {
+				return fmt.Errorf("update Argus instance: %w", err)
+			}
+			instanceId := model.InstanceId
+
+			// Wait for async operation, if async mode not enabled
+			if !model.Async {
+				s := spinner.New(cmd)
+				s.Start("Updating instance")
+				_, err = wait.UpdateInstanceWaitHandler(ctx, apiClient, instanceId, model.ProjectId).WaitWithContext(ctx)
+				if err != nil {
+					return fmt.Errorf("wait for Argus instance update: %w", err)
+				}
+				s.Stop()
+			}
+
+			operationState := "Updated"
+			if model.Async {
+				operationState = "Triggered update of"
+			}
+			cmd.Printf("%s instance %q\n", operationState, instanceLabel)
+			return nil
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), planIdFlag, "Plan ID")
+	cmd.Flags().String(planNameFlag, "", "Plan name")
+	cmd.Flags().String(instanceNameFlag, "", "Instance name")
+}
+
+func parseInput(cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	instanceId := inputArgs[0]
+
+	globalFlags := globalflags.Parse(cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &cliErr.ProjectIdError{}
+	}
+
+	planId := flags.FlagToStringPointer(cmd, planIdFlag)
+	planName := flags.FlagToStringValue(cmd, planNameFlag)
+	instanceName := flags.FlagToStringPointer(cmd, instanceNameFlag)
+
+	if planId != nil && (planName != "") {
+		return nil, &cliErr.ArgusInputPlanError{
+			Cmd: cmd,
+		}
+	}
+
+	if planId == nil && planName == "" {
+		if instanceName == nil {
+			return nil, &cliErr.EmptyUpdateError{}
+		} else if *instanceName == "" {
+			return nil, fmt.Errorf("new instance name cannot be empty.")
+		}
+	}
+
+	return &inputModel{
+		GlobalFlagModel: globalFlags,
+		InstanceId:      instanceId,
+		PlanId:          planId,
+		PlanName:        planName,
+		InstanceName:    instanceName,
+	}, nil
+}
+
+type argusClient interface {
+	UpdateInstance(ctx context.Context, instanceId, projectId string) argus.ApiUpdateInstanceRequest
+	ListPlansExecute(ctx context.Context, projectId string) (*argus.PlansResponse, error)
+	GetInstanceExecute(ctx context.Context, instanceId, projectId string) (*argus.GetInstanceResponse, error)
```

Comment:
Currently this is not used, right? But might be if we decide to call this directly instead of `GetInstanceDetails`